website/addons/gitlab/static/gitlabFangornConfig.js

```diff
@@ -0,0 +1,387 @@
+                    m('p', 'Some of these files can\'t be deleted but you can delete the ones highlighted with green. This action is irreversible.'),
+                    deleteList.map(function(n){
+                        return m('.fangorn-canDelete.text-success', n.data.name);
+                    }),
+                    noDeleteList.map(function(n){
+                        return m('.fangorn-noDelete.text-warning', n.data.name);
+                    })
+                ]);
+            mithrilButtonsMultiple =  m('div', [
+                    m('span.btn.btn-default', { 'class' : 'text-default', onclick : function() { cancelDelete(); } }, 'Cancel'),
+                    m('span.btn.btn-danger', { 'class' : 'text-danger', onclick : function() { runDeleteMultiple.call(tb, deleteList); }  }, 'Delete Some')
+                ]);
+        }
+        tb.modal.update(mithrilContentMultiple, mithrilButtonsMultiple, m('h3.break-word.modal-title', 'Delete multiple files?'));
+    }
+
+    return true; // Let fangorn know this config option was used.
+}
+
+
+// Define Fangorn Button Actions
+var _gitlabItemButtons = {
+    view: function (ctrl, args, children) {
+        var tb = args.treebeard;
+        var item = args.item;
+        var buttons = [];
+        function _downloadEvent(event, item, col) {
+            event.stopPropagation();
+            var branch = tem.data.extra.ref || item.data.extra.fileSha;
+            window.location = waterbutler.buildTreeBeardDownload(item, {branch: branch});
+        }
+        // Download Zip File
+        if (item.kind === 'folder') {
+            var branchArray = [];
+            if (item.data.branches) {
+                item.data.branch = item.data.extra.ref || item.data.extra.fileSha;
+                for (var i = 0; i < item.data.branches.length; i++) {
+                    var selected = item.data.branches[i] === item.data.branch ? 'selected' : '';
+                    branchArray.push(m('option', {
+                        selected: selected,
+                        value: item.data.branches[i]
+                    }, item.data.branches[i]));
+                }
+            }
+            if (item.data.addonFullname) {
+                buttons.push(
+                    m.component(Fangorn.Components.dropdown, {
+                        'label': 'Branch: ',
+                        onchange: function (event) {
+                            changeBranch.call(tb, item, event.target.value);
+                        },
+                        icon: 'fa fa-external-link',
+                        className: 'text-info'
+                    }, branchArray)
+                );
+            }
+            if (tb.options.placement !== 'fileview') {
+                // If File and FileRead are not defined dropzone is not supported and neither is uploads
+                if (window.File && window.FileReader && item.data.permissions && item.data.permissions.edit) {
```

Comment:
The 'Upload', 'Create Folder', and 'Delete' buttons can be removed from the read-only addon.